pkgs/development/python-modules/transmission-rpc/default.nix

```diff
@@ -6,18 +6,35 @@
 , requests
 , yarl
 , pythonOlder
+, fetchFromGitHub
+, poetry-core
 }:
 
 buildPythonPackage rec {
   pname = "transmission-rpc";
   version = "3.3.0";
   disabled = pythonOlder "3.6";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "ef3a931fc1f1db74edf8660e475b9295e0904ee922030ef0e45b0c73f4be65ae";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+      owner = "Trim21";
+      repo = "transmission-rpc";
+      rev = "v${version}";
+      sha256 = "sha256-Ys9trQMCHqxBSaTobWt8WZwi1F8HKTUKaIxvyo6ZPP0=";
   };
 
+  # remove once upstream has tagged version with dumped typing-extensions
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'version = "3.10.0.2"' 'version = "4.0.1"' \
```

Comment:
If we don't update the package version and just apply a patch there is no need to patch the version at all.

Reply:
I am unsure what you mean here?

Reply:
The `version` stays at `3.3.0`, we are only relaxing `typing_extensions` to make this work. So lets not update the version in `pyproject.toml` unless upstream released a new version and forgot to do it.

Reply:
So it should look like

```
  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace 'typing_extensions = ">=3.7.4.2,<4.0.0.0"' 'typing_extensions = "*"'
  '';
  ```

 instead?

Reply:
Fixed in latest release @mweinelt. Didn't see it as I don't get notifications about reactions :D

Reply:
My bad, sorry!